internal/requests/username_password_request_test.go

```diff
@@ -8,57 +8,85 @@ import (
 	"testing"
 
 	"github.com/AzureAD/microsoft-authentication-library-for-go/internal/msalbase"
-	"github.com/AzureAD/microsoft-authentication-library-for-go/internal/wstrust"
 )
 
-var testUPAuthorityInfo, _ = msalbase.CreateAuthorityInfoFromAuthorityURI("https://login.microsoftonline.com/v2.0/", true)
-var testUPAuthParams = msalbase.CreateAuthParametersInternal("clientID", testUPAuthorityInfo)
-var upWRM = new(MockWebRequestManager)
-var usernamePassRequest = &UsernamePasswordRequest{
-	authParameters: testUPAuthParams,
-}
-var tdr = &TenantDiscoveryResponse{
-	AuthorizationEndpoint: "https://login.microsoftonline.com/v2.0/authorize",
-	TokenEndpoint:         "https://login.microsoftonline.com/v2.0/token",
-	Issuer:                "https://login.microsoftonline.com/v2.0",
-}
-var managedUserRealm = &msalbase.UserRealm{
-	AccountType: "Managed",
-}
-var errorUserRealm = &msalbase.UserRealm{
-	AccountType: "",
-}
-var federatedUserRealm = &msalbase.UserRealm{
-	AccountType:           "Federated",
-	FederationMetadataURL: "fedMetaURL",
+// TODO(jdoak): Replace these tests with subtests to eliminate these globals
+// or table driven.
+
+var (
+	testUPAuthorityInfo msalbase.AuthorityInfo
+	uprTestAuthParams   msalbase.AuthParametersInternal
+
+	upWRM               = new(MockWebRequestManager)
+	usernamePassRequest *UsernamePasswordRequest
+
+	managedUserRealm = msalbase.UserRealm{
+		AccountType: "Managed",
+	}
+	errorUserRealm = msalbase.UserRealm{
+		AccountType: "",
+	}
+	federatedUserRealm = msalbase.UserRealm{
+		AccountType:           "Federated",
+		FederationMetadataURL: "fedMetaURL",
+	}
+)
+
+func init() {
+	var err error
+	testUPAuthorityInfo, err = msalbase.CreateAuthorityInfoFromAuthorityURI("https://login.microsoftonline.com/v2.0/", true)
+	if err != nil {
+		panic(err)
+	}
+
+	uprTestAuthParams = msalbase.CreateAuthParametersInternal("clientID", createTestAuthorityInfo())
+	uprTestAuthParams.Endpoints = msalbase.CreateAuthorityEndpoints(
+		"https://login.microsoftonline.com/v2.0/authorize",
+		"https://login.microsoftonline.com/v2.0/token",
+		"https://login.microsoftonline.com/v2.0",
+		"login.microsoftonline.com",
+	)
+
+	usernamePassRequest = &UsernamePasswordRequest{
+		authParameters: uprTestAuthParams,
+	}
 }
 
+// TODO(msal expert): This test SEEMS borked. .Execute()
```

Comment:
This flow should work, we can find a "federated" test user to try with. Maybe log an issue for this test to be fixed? @keegan-caruso - can you help with this?

Reply:
Going to let this finish offline, there are bugs to review all tests.  TODO will remain for the moment.